lightning_client.go

```diff
@@ -1091,6 +1097,75 @@ type CustomMessage struct {
 	Data []byte
 }
 
+// CustomMessageFilter provides a filter for subscribing to custom messages.
+type CustomMessageFilter struct {
```

Comment:
This is a nice abstraction, but not sure if we should put this in `lndclient` as it's more application related imo. Some users may create positive filters some may create negative filters, etc.

Reply:
Good point, I was on the fence about it too. Looking at this again, if we wanted a filter it could be a `FilterMessage(CustomMessage)` closure that you can pass in, rather than fixing the structure in here. 

Less code/more good, so will remove!

Reply:
Yeah a closure could work just fine. Avoids waking up whoever is waiting on the channel and gives more opportunity for customization.